common/testing/testlogger/testlogger.go

```diff
@@ -206,31 +219,43 @@ func NewTestLogger(t TestingT, mode Mode, opts ...LoggerOption) *TestLogger {
 			t:               t,
 			logExpectations: false,
 			level:           zapcore.DebugLevel,
+			logCaller:       true,
 			mode:            mode,
 		},
 	}
-	tl.state.panicOnError.Store(true)
-	tl.state.panicOnDPanic.Store(true)
-	tl.state.mu.expectations = map[Level]*list.List{
-		Debug: list.New(),
-		Info:  list.New(),
-		Warn:  list.New(),
-		Error: list.New(),
-	}
+	tl.state.mu.expectations = make(map[Level]*list.List)
+	tl.state.failOnError.Store(true)
+	tl.state.failOnDPanic.Store(true)
```

Comment:
Should this we also add `tl.state.failOnFatal.Store(true)`? Or if not, explain why not.

Reply:
We should; this was an oversight when translating between the two implementations.